src/main/java/org/xerial/snappy/SnappyOutputStream.java

```diff
@@ -102,7 +108,9 @@ public SnappyOutputStream(OutputStream out, int blockSize, BufferAllocatorFactor
         inputBuffer = inputBufferAllocator.allocate(inputSize);
         outputBuffer = outputBufferAllocator.allocate(outputSize);
 
-        outputCursor = SnappyCodec.currentHeader.writeHeader(outputBuffer, 0);
+        if (writeHeader) {
```

Comment:
Can we implement this by function overload?  For example, defining a protected method (writeHeader()), then override this in Hadoop stream side.